template_tag.go

```diff
@@ -0,0 +1,84 @@
+package faker
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+	"text/template"
+)
+
+// EvaluateTemplateFields evaluates the template-tagged fields for a struct value `v` of type `t`.
+// tagName is the field tag name to use (typically opts.TagName).
+func EvaluateTemplateFields(t reflect.Type, v reflect.Value, templateFields []int, tagName string) error {
+	if len(templateFields) == 0 {
+		return nil
+	}
+	// build context map from current struct values
+	ctx := make(map[string]interface{})
```

Comment:
We can modernize by using `any`
```suggestion
	ctx := make(map[string]any)
```